src/shmem_comm.h

```diff
@@ -101,6 +103,42 @@ shmem_internal_put_nb(shmem_ctx_t ctx, void *target, const void *source, size_t
     }
 }
 
+static inline
+void
+shmem_internal_put_signal_nbi(shmem_ctx_t ctx, void *target, const void *source, size_t len, 
+                              uint64_t *sig_addr, uint64_t signal, int pe)
+{
+    int node_rank;
+
+    if (len == 0) {
+        shmem_internal_put_scalar(ctx, sig_addr, &signal, sizeof(uint64_t), pe);
+        return;
+    }
```

Comment:
Code looks good.  Just wanted to suggest double checking the proposal that this is the right way to handle the `len == 0` case.

Reply:
There is no specific text added in the proposal for this case. However, in the discussion, it is mentioned that sending the signal is the expected behavior.